cpp/src/transform/row_bit_count.cu

```diff
@@ -416,8 +418,8 @@ CUDF_KERNEL void compute_row_sizes(device_span<column_device_view const> cols,
   row_span* my_branch_stack = thread_branch_stacks + (threadIdx.x * max_branch_depth);
   size_type branch_depth{0};
 
-  // current row span - always starts at 1 row.
-  row_span cur_span{tid, tid + 1};
+  // current row span - always starts at segment_length rows.
+  row_span cur_span{tid * segment_length, (tid + 1) * segment_length};
```

Comment:
I think this doesn't handle the case where there's not enough rows left to fill a segment.  Ie, we've got 25 rows and segment size 8.  The last thread is going to attempt to look at rows 24 through 32.